src/main/java/com/pnu/dev/shpaltaif/controller/AuthController.java

```diff
@@ -14,13 +14,23 @@
     @GetMapping("/login")
     public String login(Model model, String error, String logout) {
 
-        if (nonNull(error))
-            model.addAttribute(FLASH_MESSAGE_ERROR, "Неправильні ім'я користувача або пароль!");
+        if (nonNull(error)) {
+            model.addAttribute(FLASH_MESSAGE_ERROR, resolveErrorMessage(error));
+        }
 
         if (nonNull(logout))
             model.addAttribute(FLASH_MESSAGE_SUCCESS, "Ви успішно вийшли!");
 
         return "/admin/login";
     }
 
+    private String resolveErrorMessage(String error) {
+        String errorMessage = "Помилка авторизації";
+        if (error.equalsIgnoreCase("Bad credentials")) {
+            errorMessage = "Неправильні ім'я користувача або пароль!";
+        } else if (error.equalsIgnoreCase("blocked")) {
```

Comment:
I'd move this value to some enum, because it's not clear where it's coming from, especially when this literal is used in two different classes.